sw/otbn/crypto/p256_ecdsa.s

```diff
@@ -111,8 +111,9 @@ ecdsa_sign:
  * @param[out] dmem[x_r]: dmem buffer for reduced affine x_r-coordinate (x_1)
  */
 ecdsa_verify:
-  /* Validate the public key. */
-  jal      x1, check_public_key_valid
+  /* Validate the public key (jumps back here if successful). */
+  jal      x0, check_public_key_valid
```

Comment:
Wouldn't it make sense to also put the `check_public_key_valid` routine into a separate file (like `p256_isoncurve.s`)? Because currently we have `check_public_key_valid` implemented twice in `p256_ecdsa.s` as well as in `p256_ecdh.s`. This probably won't save us additional IMEM, but could prevent us from forgetting to adapt one of the implementations to potential future changes in `p256_isoncurve`.

Reply:
Personally, I would prefer it to put it into a separate file. That makes it clearer from a programmer's perspective. 
What if we would write the routine like this at the end? Then it could be callable like a normal function. 
To prevent skipping the whole function call via FI, we could then add additional `unimp` lines after the jump to the function. Or does that blow up the code too much?
```
  csrrs     x2, 0x7c0, x0
  srli      x2, x2, 3
  andi      x2, x2, 1
  bne       x2, x0, _pk_valid
  unimp
  unimp
  unimp
  
  _pk_valid:
  ret
```